src/transfer-manager.ts

```diff
@@ -258,7 +260,7 @@ export class TransferManager {
         {},
         options.passthroughOptions
       );
-      if (options.prefix) {
+      if (options.prefix || passThroughOptionsCopy.destination) {
```

Comment:
having iterated on a few approaches to get to this point, I've realized now that (in hindsight), `options.prefix` and `options.passthroughoptions.destination` fulfill the same purpose in `TransferManager`. Both are being prefixed toe the `file.name` via `path.join`. Technically, we only need one of the two.

That being said, I think we should probably keep `options.prefix` for:
- backwards compatibility for users that are currently using it
- users that prefer to set the more visible `options.prefix` over setting `options.passthroughoptions.destination`

Reply:
Yes, I think for the time let's keep both as to avoid a breaking change. I will make a note to clean this up when we prepare the next major revision.